nucleus/payara-modules/healthcheck-core/src/main/java/fish/payara/nucleus/healthcheck/preliminary/BaseHealthCheck.java

```diff
@@ -98,15 +98,15 @@
     private final AtomicInteger checksDone = new AtomicInteger();
     private final AtomicInteger checksFailed = new AtomicInteger();
     private final AtomicBoolean inProcess = new AtomicBoolean(false);
-    private volatile HealthCheckResultStatus mostRecentCumulativeStatus;
+    private volatile HealthCheckResult mostRecentCumulativeResult;
```

Comment:
The result is not _cumulative_. Its collecting entries in a list. The accumulation of the status of these entries is _cumulative_. So this should just be the `mostRecentResult`.